### PR TITLE
Add `change` event

### LINKED ISSUE #158: Does not fire `change` event on change
Should be aligned with the rest of elements.
Needed in vaadin-crud for setting correctly dirty flag.

### PR DESCRIPTION
**Pull request by samiheikki**
Fixes #158

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-select/171)

### CODE REVIEW

**Comment by jouni** [comment]
Needs documentation.

**Comment by tomivirkki** [comment]
*[a discussion](https://reviewable.io/reviews/vaadin/vaadin-select/171#-LQ3c6N39dd-gVZCyn3e:-LQ3c6N39dd-gVZCyn3f:b-2a0sw9) (no related file):*
This will dispatch a "change" event even if the dropdown menu is closed without the value being changed (outside click or esc)

---

**Comment by tomivirkki** [comment]
*[a discussion](https://reviewable.io/reviews/vaadin/vaadin-select/171#-LQDzhAv-PuWrfGCF4iF:-LQDzhAv-PuWrfGCF4iG:bf7vprn) (no related file):*
```
$0.opened = true;
$0.opened = false;
```
will emit a "change" event even though it's not user-originated.

---

**Comment by samiheikki** [comment]
Polymer 3 tests are failing on master as well: https://travis-ci.org/vaadin/vaadin-select/jobs/491185839

**Comment by samiheikki** [comment]
Tests are passing now

File: test/select-test.html [hunk 1]
```diff
@@ -800,6 +801,75 @@
...
+          menu.selected = 1;
+
+          expect(changeSpy.callCount).to.equal(0);
+        });
+
+        it('should not fire for changes when closed', () => {
+          menu.selected = 1;
+          select.opened = true;
+          select.opened = false;
+          it('should fire exactly one `change` event', () => {
```
Comment (tomivirkki) [inline test/select-test.html]: What's happening here?

File: src/vaadin-select.html [hunk 2]
```diff
@@ -474,6 +485,18 @@
...
           }
           this.validate();
+
+          // In the next _detectAndDispatchChange() call, the change detection should pass
+          this._lastCommittedValue = undefined;
+          this._detectAndDispatchChange();
+        }
+
+        _detectAndDispatchChange() {
+          if (this.value !== this._lastCommittedValue && this._lastCommittedWasOpened) {
```
Comment (tomivirkki) [inline src/vaadin-select.html]: Opened state is not a valid indicator for user-originated change after https://github.com/vaadin/vaadin-select/pull/185

File: test/select-test.html (hunk 1 above)
Comment (samiheikki) [reply test/select-test.html]: That's something extra :D Removed it.

File: src/vaadin-select.html (hunk 2 above)
Comment (samiheikki) [reply src/vaadin-select.html]: Fixed. Also added test to fire `change` event when value is changed by searching a value when input is closed

File: src/vaadin-select.html [hunk 3]
```diff
@@ -528,6 +559,7 @@
             }
             this.validate();
             window.removeEventListener('scroll', this._boundSetPosition, true);
+            this._wasOpened = true;
```
Comment (Haprog) [inline src/vaadin-select.html]: This doesn't really work well. This causes a bug that the next value change after closing always triggers a `change` event (regardless of if it was user initiated or not).

```js
select.opened = true;
select.opened = false;
```

**Comment by web-padawan** [comment]
Closing in favor of #196

**Pull request closed without merging**